spopt/tests/test_skater.py

```diff
@@ -0,0 +1,17 @@
+import numpy as np
+import libpysal
+import geopandas as gpd
+from .. import SKATER
+
+def test_skater():
```

Comment:
IMHO we should be using the `unittest`-style testing (as in [`esda`](https://github.com/pysal/esda/blob/e66098341ca8c8b31ceff2025edeefbf10df023f/esda/tests/test_moran.py#L12)), but I will defer to @sjsrey's judgment.

Reply:
Thanks for pointing this out. I just followed the format of [test_maxp.py](https://github.com/pysal/spopt/blob/master/spopt/tests/test_maxp.py) and [test_region_k_means.py](https://github.com/pysal/spopt/blob/master/spopt/tests/test_region_k_means.py). I have not written this before. Do you know whether there are the unittest -style testing for these two methods? How could they pass the checks?  @jGaboardi

Reply:
@xf37 I don't mean to suggest that the way you wrote the test is wrong. It is a perfectly valid test. When I say "`unittest`-style", I simply mean with the `unittest` package in something like the format below. By creating a `TestSkater(unittest.TestCase)` instance and adding the `Skater` model as an attribute in `set_up()` we can reuse the model in more tests. In the current test we are only testing the values of `model.labels_`, but there are many of things we could (should?) be testing.

```python
import unittest

class TestSkater(unittest.TestCase):

    def set_up():
        self.pth = libpysal.examples.get_path("mexicojoin.shp")
        self.mexico = gpd.read_file(self.pth)
        self.mexico["count"] = 1
        self.attrs_name = [f"PCGDP{year}" for year in range(1950, 2010, 10)]
        self.w = libpysal.weights.Queen.from_dataframe(self.mexico)
        np.random.seed(123456)
        self.model = Skater(gdf=mexico, w=w, attrs_name=attrs_name)
        self.model.solve()

    def test_labels_(self):
        observed_labels = self.model.labels_
        known_labels = np.array(
            [
                0,
                0,
                1,
                2,
                2,
                1,
                1,
                1,
                1,
                1,
                3,
                2,
                1,
                1,
                1,
                1,
                4,
                1,
                1,
                1,
                1,
                1,
                0,
                0,
                0,
                1,
                1,
                1,
                1,
                0,
                1,
                1,
            ]
     )
     numpy.testing.assert_equal(observed_labels, known_labels)
```

Reply:
Gotcha, thank you for explaining.